src/sentry/issues/forecasts.py

```diff
@@ -51,6 +53,23 @@ def generate_and_save_forecasts(groups: Sequence[Group]) -> None:
     Generates and saves a list of forecasted values for each group.
     `groups`: Sequence of groups to be forecasted
     """
+
+    # Check if any of the groups are source groups involved in an unmerge in the last 24 hrs
+    # If a group was a source group in an unmerge in the last 24 hr, then we must query all the
+    # groups involved in the unmerge to get accurate event counts due to a Clickhouse bug.
+    if groups:
+        org_id = groups[0].project.organization.id
+        unmerged_keys = [
+            key.partition("unmerged-groups")[1] + key.partition("unmerged-groups")[2]
+            for key in cache._cache.keys()
+            if "unmerged-groups:" + str(org_id) + ":" in key
+        ]
+        for key in unmerged_keys:
+            unmerge_groups_ids = cache.get(key)
+            if unmerge_groups_ids:
+                unmerge_groups = Group.objects.filter(id__in=unmerge_groups_ids)
```

Comment:
try to prevent cross organization operations from being possible: `project__organization_id=organization.id`